src/pages/NewsList/NewsList.tsx

```diff
@@ -46,37 +58,60 @@ export default function NewsList() {
 			return titleMatches || summaryMatches || authorMatches;
 		});
 	}
+	if (selectedTopics.length > 0) {
+		filteredData = filteredData.filter(
+			(newsItem: NewsData) => newsItem.topics.slice(0, 3).some(
+				(topicData) => selectedTopics.includes(topicData.topic),
+			),
+		);
+	}
 
 	return (
 		<div className="container mt-24 mx-auto px-7">
-			<h1 className="text-2xl font-bold mb-7">News & Sentiment Trending for Apple</h1>
+			<h1 className="text-2xl font-semibold mb-7">News & Sentiment Trending for Apple</h1>
 
 			{loading ? (
 				<div className="container mx-auto">
 					<h2>Loading news...</h2>
 				</div>
 			) : (
 				<>
-					<div className="mb-4">
+					<div className="flex justify-between flex-wrap gap-4 mb-4 items-end">
 						<input
 							type="search"
 							aria-label="search"
 							id="search"
 							placeholder="Search..."
 							value={searchText}
 							onChange={(e) => setSearchText(e.target.value)}
-							className="border border-gray-300 rounded-lg px-4 py-2"
+							className="border border-gray-500 rounded-lg px-4 py-2"
 						/>
+
+						<div className="flex flex-wrap gap-2">
+							{Object.keys(topicColors).map((topic) => (
+								<button
+									type="button"
+									key={topic}
+									onClick={() => handleTopicClick(topic)}
+									className={`flex items-center px-2 py-1 text-xs text-white rounded-lg cursor-pointer transition-colors ${
+										selectedTopics.includes(topic) ? 'bg-gray-600 border border-gray-600' : 'bg-transparent border border-gray-500'
```

Comment:
oh, this is hard to read. Maybe I would move the condition before render of this button.

Reply:
and made it multiline